src/assets/wise5/authoringTool/node/nodeAuthoringComponent.ts

```diff
@@ -152,18 +152,29 @@ class NodeAuthoringController {
   }
 
   previewStepInNewWindow() {
-    const data = { constraints: true };
-    this.saveEvent('stepPreviewed', 'Navigation', data);
-    window.open(`${this.ConfigService.getConfigParam('previewProjectURL')}/${this.nodeId}`);
+    this.savePreviewEventAndOpenPreview(true);
   }
 
   previewStepWithoutConstraintsInNewWindow() {
-    const data = { constraints: false };
+    this.savePreviewEventAndOpenPreview(false);
+  }
+
+  private savePreviewEventAndOpenPreview(constraints: boolean) {
+    this.saveStepPreviewedEvent(constraints);
+    window.open(this.createPreviewURL(this.nodeId, constraints));
+  }
```

Comment:
Change this function to protected (Angular 14 new feature that works with calls from templates), and replace calls (in the template) to ```previewStepWithoutConstraintsInNewWindow()``` and ```previewStepInNewWindow()``` to directly call this function? Is there a need to have those two functions if all they do is call this function?

```suggestion
  protected savePreviewEventAndOpenPreview(constraints: boolean): void {
    this.saveStepPreviewedEvent(constraints);
    window.open(this.createPreviewURL(this.nodeId, constraints));
  }
```